aten/CMakeLists.txt

```diff
@@ -123,15 +123,18 @@ IF (APPLE AND CMAKE_COMPILER_IS_GNUCC)
   ENDIF ()
 ENDIF ()
 
-IF (WITH_OPENMP)
+IF (WITH_OPENMP AND NOT OPENMP_FOUND)
   FIND_PACKAGE(OpenMP)
-  IF(OPENMP_FOUND)
-    MESSAGE(STATUS "Compiling with OpenMP support")
-    SET(CMAKE_C_FLAGS "${CMAKE_C_FLAGS} ${OpenMP_C_FLAGS}")
-    SET(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} ${OpenMP_CXX_FLAGS}")
-    SET(CMAKE_EXE_LINKER_FLAGS "${CMAKE_EXE_LINKER_FLAGS} ${OpenMP_EXE_LINKER_FLAGS}")
-  ENDIF(OPENMP_FOUND)
-ENDIF (WITH_OPENMP)
+ENDIF (WITH_OPENMP AND NOT OPENMP_FOUND)
+SET(OPENMP_FOUND ON CACHE BOOL "OpenMP Support found")
+
+IF(OPENMP_FOUND)
+  MESSAGE(STATUS "Compiling with OpenMP support")
+  SET(CMAKE_C_FLAGS "${CMAKE_C_FLAGS} ${OpenMP_C_FLAGS}")
```

Comment:
Will these flags variables be available even if you don't cache them explicitly?

Reply:
good catch, probably not. will fix.